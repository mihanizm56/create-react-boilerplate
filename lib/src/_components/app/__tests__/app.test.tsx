import React from 'react';
import { shallow } from 'enzyme';
import { App } from '../index';

jest.mock('@wildberries/service-router', () => ({
  RouteNode: 'RouteNode',
  RouterProvider: 'RouterProvider',
}));

jest.mock('@/_components/root-components', () => ({
  RootComponents: 'RootComponents',
}));

const testProps: any = {
  router: jest.fn().mockName('router'),
};

describe('App component', () => {
  describe('render', () => {
    let component;

    beforeEach(() => {
      component = shallow(<App {...testProps} />);//eslint-disable-line
    });

    test('render with all props', () => {
      expect(component).toMatchSnapshot();
    });

    test('render RootComponents', () => {
      expect(component.find('RootComponents').length).toEqual(1);
    });

    test('render RouterProvider', () => {
      expect(component.find('RouterProvider').length).toEqual(1);
    });

    // must not use "error" text in test name due to cli is catching "bad" logs
    test('render RouteNode if there is no errror', () => {
      const customProps = { ...testProps, isAppNetError: false };
      const customComponent = shallow(<App {...customProps} />);//eslint-disable-line

      expect(customComponent.find('RouteNode').length).toEqual(1);
    });
  });
});
