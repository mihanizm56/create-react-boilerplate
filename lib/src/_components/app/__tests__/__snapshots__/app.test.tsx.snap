// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component render render with all props 1`] = `
<Fragment>
  <RootComponents />
  <RouterProvider
    key="1"
    router={[MockFunction router]}
  >
    <RouteNode
      nodeName=""
    >
      <Component />
    </RouteNode>
  </RouterProvider>
</Fragment>
`;
